### ISSUE TITLE
Locating objects without bounding boxes

### ISSUE THREAD

**Issue by qtreh** [qtreh/papers#4]
Javier Ribera, David Güera, Yuhao Chen, Edward J. Delp
CVPR 2019
[arXiv:1806.07564 [cs.CV]](https://arxiv.org/abs/1806.07564)

**Goal**

Most recent advances in object detection involve a bounding box. This helps delimiting the area of interest in an intuitive way. But the annotation process of bounding boxes is tedious, especially in video tracking, whereas single points are way faster to draw. This is sufficient in numerous practical use cases, such as object counting. They also happen to be better suited to densely packed situations with crowded, overlapping instances.

Bounding boxes are usually good at detecting a variety of different objects. Instance detection is usually domain-specific. Recent approaches 

in crowd counting use a density map around particular traits of people (only the head most of the time). Precision agriculture and pupil tracking are two other examples of specific problems in which a bounding box is either useless or even degrading overall performances.

![nobox1](https://user-images.githubusercontent.com/8530703/82154492-b0e76280-986e-11ea-8da8-b3b55d7bb5e4.png)

The authors used a measure of distance between two sets of points, called the Hausdorff distance. Used in very specific domains such as 3D scene reconstruction, segmentation quality evaluation of face recognition, they adapt it in a fully convolutional network to perform standard object detection, with great success.

**Difficulties - The Hausdorff distance** 

The Hausdorff distance, in a discretized and bounded context, measures the distance between two sets of points by taking the maximum distance of all the distances from a point in one set to the closest point in the other set. 

![hausdorff](https://user-images.githubusercontent.com/8530703/82154571-14719000-986f-11ea-98a7-518929840ee9.png)

As this metric is very sensitive to outliers, the Average Hausdorff distance is more commonly used.

The goal is to use this metric as a loss function in a CNN. To be used, it still needs to embed the pixel coordinates (needed when backpropagating), and the estimated number of instances of the image.

**Contributions**

Introducing the Weighted Hausdorff Distance:

![hausdorff_formula](https://user-images.githubusercontent.com/8530703/82154605-4f73c380-986f-11ea-88a4-65ef9ed92351.png)

![hausdorff_formula_2](https://user-images.githubusercontent.com/8530703/82154623-67e3de00-986f-11ea-98ac-db2028751b76.png)

This new formulation enforces the following points. px marks the presence or not of an instance, between 0 and 1. Multiplying by px in the first term penalizes activated areas where there is no groundtruth point nearby. In the second term, if there are two activations near a groundtruth point, the lower one will be highly penalized. 

This method is better than a pixelwise loss, in which we would only match two points if they are the same. Training is impossible as we have no information if the points are closed to each other.

The whole training loss includes an L1 loss for the object count. The number of instances is estimated using a Beta mixture model-based thresholding. 

**Results**

The authors used three datasets, with pupil tracking, field crop detection, and crowd counting in the mall. That third dataset is their own and has been made public. 

Benchmarks show that the performance of localization is way above the usual state-of-the-art.

As for counting tasks, it achieves comparable results against the domain-specific systems.

A PyTorch implementation is available. One drawback, that the authors plan on working on, is that this object detection framework only detects on a single category of instance.
